src/main/java/org/commcare/session/RemoteQuerySessionManager.java

```diff
@@ -286,6 +286,13 @@ private void validateUserAnswers() {
             if (!StringUtils.isEmpty(value) && queryPrompt.isInvalidInput(new EvaluationContext(ec, currentRef))) {
                 errors.put(key, queryPrompt.getValidationMessage(ec));
             }
+            if (StringUtils.isEmpty(value) && queryPrompt.isRequired(ec)) {
+                String message = queryPrompt.getRequiredMessage(ec);
+                if (message.replace('\u00A0',' ').isBlank()) {
```

Comment:
This replaces the [Non-breaking space](https://www.ascii-code.com/CP1252/160) that is returned from `getRequiredMessage` for fields that are required but don't have a `Require Message` set in app manager. I was stuck on this for a while since `" "`(Space) and `" "`(Non-breaking space) appear the same, but calling `isBlank` or `isEmpty` on a Non-breaking Space returns false. I'm curious if there is a reason a Non-breaking space is returned instead of `null` or an empty string?

Reply:
Are you sure that the nbsp is not getting set in HQ for your app ?

Reply:
> Are you sure that the nbsp is not getting set in HQ for your app ?

Looking at this again, I do think the nbsp is coming from HQ. I haven't tracked down exactly where, but it seems to be coming from Backbone js.

Reply:
Think this needs to be fixed on HQ in that case instead of us trying to do any string manipulation here.

Reply:
It turns out that nbsp is actually intentional. It's coming from this translation method: [dumps](https://github.com/dimagi/commcare-translations/blob/0de322558e2bd74b70f051c8790257b0f5929cbe/commcare_translations.py#L107) that is part of generating app strings. Does it make sense in that case to keep `replace` call?

Reply:
I see, I am not sure why we are doing that on HQ but I would avoid correcting for HQ bheaviour on FP. I think in context of this PR, we can just assign `message` as `queryPrompt.getRequiredMessage(ec)` and remove this handling. If this becomes a problem down the line, we can push for correcting the behaviour on HQ. I think the right behaviour here will be for HQ to assing the default message instead of nbsp to the string.